operator/pkg/lib/sidecars/restart/restart_test.go

```diff
@@ -0,0 +1,304 @@
+package restart_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/kyma-project/istio/operator/pkg/lib/sidecars/restart"
+	"github.com/stretchr/testify/require"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+const annotationName = "kubectl.kubernetes.io/restartedAt"
+
+func TestRestart(t *testing.T) {
+	ctx := context.TODO()
+
+	t.Run("should return warning when pod has no owner", func(t *testing.T) {
+		// given
+		c := fakeClient(t)
+
+		podList := v1.PodList{
+			Items: []v1.Pod{
+				podWithoutOwnerFixture("p1", "test-ns"),
+			},
+		}
+
+		// when
+		warnings, err := restart.Restart(ctx, c, podList)
+
+		// then
+		require.NoError(t, err)
+		require.NotEmpty(t, warnings)
+
+		require.Equal(t, "p1", warnings[0].Name)
+		require.Contains(t, warnings[0].Message, "OwnerReferences was not found")
+	})
+
+	t.Run("should return warning when pod is owned by a Job", func(t *testing.T) {
+		// given
+		c := fakeClient(t)
+
+		podList := v1.PodList{
+			Items: []v1.Pod{
+				podFixture("p1", "test-ns", "Job", "owningJob"),
+			},
+		}
+
+		// when
+		warnings, err := restart.Restart(ctx, c, podList)
+
+		// then
+		require.NoError(t, err)
+		require.NotEmpty(t, warnings)
+
+		require.Equal(t, "p1", warnings[0].Name)
+		require.Contains(t, warnings[0].Message, "owned by a Job")
+	})
+
+	t.Run("should return warning when pod is owned by a Job", func(t *testing.T) {
+		// given
+		c := fakeClient(t)
+
+		podList := v1.PodList{
+			Items: []v1.Pod{
+				podFixture("p1", "test-ns", "Job", "owningJob"),
+			},
+		}
+
+		// when
+		warnings, err := restart.Restart(ctx, c, podList)
+
+		// then
+		require.NoError(t, err)
+		require.NotEmpty(t, warnings)
+
+		require.Equal(t, "p1", warnings[0].Name)
+		require.Contains(t, warnings[0].Message, "owned by a Job")
+	})
+
+	t.Run("should rollout restart Deployment if the pod is owned by one", func(t *testing.T) {
+		// given
+		c := fakeClient(t, &appsv1.Deployment{ObjectMeta: metav1.ObjectMeta{Name: "owner", Namespace: "test-ns"}})
+
+		podList := v1.PodList{
+			Items: []v1.Pod{
+				podFixture("p1", "test-ns", "Deployment", "owner"),
+			},
+		}
+
+		// when
+		warnings, err := restart.Restart(ctx, c, podList)
+
+		// then
+		require.NoError(t, err)
+		require.Empty(t, warnings)
+
+		obj := appsv1.Deployment{}
+		err = c.Get(context.TODO(), types.NamespacedName{Namespace: "test-ns", Name: "owner"}, &obj)
+		require.NoError(t, err)
+
+		require.NotEmpty(t, "owner", obj.Annotations[annotationName])
```

Comment:
I think the asserts of the annotation don't work as expected. As the function signature is `NotEmpty(t TestingT, object interface{}, msgAndArgs ...interface{})`. That means in this case `owner` is the tested object and `obj.Annotations[annotationName]` is passed as `msgAndArgs`.
We might also use the `ResourceVersion` of the object to assert if it was changed.